docs/arch/controller.md

```diff
@@ -0,0 +1,40 @@
+# Architecture Document for Controller Component
+
+```mermaid
+sequenceDiagram
+    actor User
+    User->>Controller: sends pipeline jobs as YAML and repository link
+    alt is request malformed
+        Controller-->>User: Nok
+    else is well
+        Controller-->>User: Ok
+    end
+    Controller->>Database: saves pipeline in database
+    loop over steps
+        Controller-->>Scheduler: sends step (over GRPC)
+        Scheduler-->>Controller: job succeeded or not
+    end
+    User->>Controller: get pipeline state
+```
+
+## Glossary
+- action : list of steps, a declarative file can only contain one action.
+- step : a step is the most atomic component of our pipeline.
```

Comment:
You should be more precise in what you could put in a step

Reply:
I second this

Reply:
I third this, we are going to add an example of manifest describing a basic pipeline